### ISSUE TITLE
IP addresses change does not trigger update in browser

### ISSUE THREAD

**Issue by PhilippSelenium** [jstasiak/python-zeroconf#235]
# Steps to reproduce
1. Use example code from README:

```
from zeroconf import ServiceBrowser, Zeroconf


class MyListener:

    def remove_service(self, zeroconf, type, name):
        print("Service %s removed" % (name,))

    def add_service(self, zeroconf, type, name):
        info = zeroconf.get_service_info(type, name)
        print("Service %s added, service info: %s" % (name, info))

    def update_service(self, zeroconf, type, name):
        info = zeroconf.get_service_info(type, name)
        print("Service %s updated, service info: %s" % (name, info))

zeroconf = Zeroconf()
listener = MyListener()
browser = ServiceBrowser(zeroconf, "_http._tcp.local.", listener)
try:
    input("Press enter to exit...\n\n")
finally:
    zeroconf.close()
```
2. Change IP address
exp: An update message is printed.
act: Nothing is printed

# Notes:
1. I can see the packets sent in wireshark.
2. It works with version 0.24.4

zeroconf version: 0.24.5
python version: 3.6.7

**Comment by jstasiak** [comment]
I narrowed this down to https://github.com/jstasiak/python-zeroconf/pull/228, cc @mattsaxon, I'll try to see if there's something obvious that I can fix quickly.

**Comment by jstasiak** [comment]
I think this is because `ServiceBrowser.update_record()` only triggers `ServiceStateChange` on `PTR` and `TXT` record updates: https://github.com/jstasiak/python-zeroconf/blob/a79015e7c4bdc843d97bd5c82ef8ed4eeae01a34/zeroconf/__init__.py#L1428

**Comment by mattsaxon** [comment]
It’s is sort of by design that it doesn’t change with a changed IP address. We do need to change it so it triggers on SRV update though so I do appreciate this is both a regression and needs fixing, but I’d rather forward fix it to trigger on the SRV change as you suggest.

**Comment by jstasiak** [comment]
Just so I understand – is it by design in relation to any part of RFC 6762 or 6763?

**Comment by mattsaxon** [comment]
Sorry, no I meant that the code was never designed to deal with an IP address change. The older code just got ‘lucky’, presumably
because  a spurious txt update came through. We should explicitly recognise SRV changes though, so that is the right change to make.

**Comment by mattsaxon** [comment]
Assign this and the other regression to
me and I’ll sort them out when I get a chance

**Comment by emontnemery** [comment]
Since this is a regression from #228, can #228 be reverted until we handle IP address change?

**Comment by mattsaxon** [comment]
I should have forward fix available by end of today

**Comment by mattsaxon** [comment]
There is a possibility of this being a breaking change as UpdateService used to get called on service addition. I feel this is wrong and was a mistake.

**Comment by jstasiak** [comment]
Yeah I noticed that too, I agree this seems wrong and I'd say let's fix this.

**Comment by jstasiak** [comment]
Fix released in version 0.26.0.
